src/Context/themeContext.js

```diff
@@ -1,24 +1,42 @@
-import { createContext, useState } from "react";
+import { createContext, useState, useEffect, useMemo } from "react";
 
-export const ThemeContext = createContext()
+export const ThemeContext = createContext();
 
-//Theme modes 
-const light = {mode: 'light', color: 'black', bg: 'white'}
-const dark = {mode: 'dark', color: 'white', bg: '#020300'}
+// Theme modes
+const light = { mode: "light", color: "black", bg: "white" };
+const dark = { mode: "dark", color: "white", bg: "#020300" };
 
-//Get the system theme
-let systemTheme = {}
-window.matchMedia('(prefers-color-scheme: dark)').matches ? systemTheme = dark : systemTheme = light
+// Get the system theme
+const getInitialTheme = () => {
+  const savedTheme = JSON.parse(localStorage.getItem("theme"));
+  if (savedTheme) {
+    return savedTheme;
+  }
+  return window.matchMedia("(prefers-color-scheme: dark)").matches
+    ? dark
+    : light;
+};
 
-export function ThemeProvider(props) {
-    const [theme, setTheme] = useState(systemTheme)
+export function ThemeProvider({ children }) {
+  const [theme, setTheme] = useState(getInitialTheme());
 
-    //Handle theme change
-    function changeTheme() {
-        theme.mode === 'light' ? setTheme(dark) : setTheme(light)
-    }
+  // Handling theme change
+  const changeTheme = () => {
+    const newTheme = theme.mode === "light" ? dark : light;
+    setTheme(newTheme);
+  };
 
-    return <ThemeContext.Provider value={{theme: theme, changeTheme: changeTheme}}>
-        {props.children}
+  // Updating localStorage whenever the theme changes
+  useEffect(() => {
+    localStorage.setItem("theme", JSON.stringify(theme));
+  }, [theme]);
+
+  // Memoizing the context value to avoid unnecessary re-renders
```

Comment:
are these comments required? they are mostly defining self explanatory code

Reply:
you are correct the comments are self explanatory , i will remove the comments